tests/by-util/test_install.rs

```diff
@@ -1764,3 +1764,18 @@ fn test_install_from_stdin() {
     assert!(at.file_exists(target));
     assert_eq!(at.read(target), test_string);
 }
+
+#[test]
+fn test_install_failing_copy_file_to_target_contain_subdir_with_same_name() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    let file = "file";
+    let dir1 = "dir1";
+
+    at.touch(file);
+    at.mkdir(dir1);
+    at.mkdir(format!("{}/{}", dir1, file));
```

Comment:
A detail: you can simplify it with `mkdir_all`.
```suggestion
    at.mkdir_all(&format!("{dir1}/{file}"));
```